vtex-sites/gatsby.store
# Feat: integrate with client-cms

Signed-off-by: Arthur Andrade <arthurfelandrade@gmail.com>## What's the purpose of this pull request?

*this PR is a refactoring of [feat: integrate with VTEX headless CMS #104](https://github.com/vtex-sites/gatsby.store/pull/104) by @danzanzini using the client-cms.

## How does it work?

This PR integrates the Faststore Starter with VTEX headless CMS using the [Client CMS Package](https://www.npmjs.com/package/@vtex/client-cms).
The integration was done in index page, where with the gatsby [getServerData function](https://www.gatsbyjs.com/docs/how-to/rendering-options/using-server-side-rendering/) all data was fetched, and the page was mounted.

Was added a src/cms/cmsHomeFallback.ts file as fallback for cases of fail in a request or for a project that don't use CMS integration.

The integration with GlobalCMS items like the Alert should be done in [other task](https://vtex-dev.atlassian.net/browse/FS-511).

## How to test it?

Open the preview link, visit the index page (/) and see if it's everything fine.

## References

[Client CMS Package](https://www.npmjs.com/package/@vtex/client-cms)
[ getServerData function](https://www.gatsbyjs.com/docs/how-to/rendering-options/using-server-side-rendering/) 

## Checklist

<em>You may erase this after checking them all ;)</em>

**PR Title and Commit Messages**

**PR Description**

**Documentation**
